scripts/screeps-private-smoke-harness.py

```
@@ -0,0 +1,509 @@
+    if plan.map_path.exists() and not plan.force_download:
+        return f"using cached map {plan.map_path}"
+    if plan.no_download:
+        return f"skipped map download; expected path is {plan.map_path}"
+    if plan.dry_run:
+        return f"would download {plan.map_url} to {plan.map_path}"
+
+    plan.maps_dir.mkdir(parents=True, exist_ok=True)
+    request = urllib.request.Request(
+        plan.map_url,
+        headers={"User-Agent": "screeps-private-smoke-harness/1.0"},
+    )
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)